KoPrivateGPT/retrieval/base.py

```diff
@@ -18,3 +23,87 @@ def ingest(self, passages: List[Passage]):
     @abstractmethod
     def retrieve_id(self, query: str, top_k: int = 5, *args, **kwargs) -> List[Union[str, UUID]]:
         pass
+
+    def fetch_data(self, ids: List[UUID]) -> List[Passage]:
+        db_origin_list = self.redis_db.get_json(ids)
+        # Sometimes redis doesn't find the id, so we need to filter that db_origin is None.
+        filter_db_origin = self.none_filter(db_origin_list)
+        # Check duplicated db instance
+        final_db_origin = self.duplicate_check(filter_db_origin)
```

Comment:
If you build duplicate_check like this, DB instance is not save to instance so everytime user calls retrieve method, same instance will create.  
Why don't you make list (which is instance variable) that save DB instance, and perform duplicate check at fetch_data?